### PR TITLE
Display new created substation in spreadsheet

### PR DESCRIPTION
**Pull request by souissimai**

### CODE REVIEW

File: src/test/java/org/gridsuite/network/map/NetworkMapControllerTest.java [hunk 1]
```diff
@@ -1276,18 +1279,19 @@ private void succeedingTestForNominalVoltages(UUID networkUuid, String variantId
 
     @Test
     public void shouldReturnSubstationsTabData() throws Exception {
+        succeedingTestForElementsInfos(NETWORK_UUID, null, ElementType.SUBSTATION, InfoType.TAB, null, resourceToString("/substations-tab-data.json"));
```
Comment (sBouzols) [inline src/test/java/org/gridsuite/network/map/NetworkMapControllerTest.java:1282]: Why ? it's exactly the same line than 1283

File: src/main/java/org/gridsuite/network/map/NetworkMapService.java [hunk 2]
```diff
@@ -53,7 +53,9 @@ private PreloadingStrategy getPreloadingStrategy(List<String> substationsIds) {
     public List<String> getSubstationsIds(UUID networkUuid, String variantId, List<Double> nominalVoltages) {
         Network network = getNetwork(networkUuid, PreloadingStrategy.COLLECTION, variantId);
         return network.getSubstationStream()
-                .filter(substation -> substation.getVoltageLevelStream().anyMatch(voltageLevel -> nominalVoltages == null || nominalVoltages.contains(voltageLevel.getNominalV())))
+                .filter(substation -> substation.getVoltageLevelStream().findAny().isEmpty() ||
+                        substation.getVoltageLevelStream().anyMatch(voltageLevel ->
+                                nominalVoltages == null || nominalVoltages.contains(voltageLevel.getNominalV())))
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/network/map/NetworkMapService.java:58]: Maybe this is better
```suggestion
                .filter(substation -> ( nominalVoltages == null && 
                                                 substation.getVoltageLevelStream().findAny().isEmpty() ) ||
                        substation.getVoltageLevelStream().anyMatch(voltageLevel ->
                                nominalVoltages == null || nominalVoltages.contains(voltageLevel.getNominalV())))
```

File: src/test/java/org/gridsuite/network/map/NetworkMapControllerTest.java [hunk 3]
```diff
@@ -1276,18 +1279,19 @@ private void succeedingTestForNominalVoltages(UUID networkUuid, String variantId
...
-        succeedingTestForElementsIds(NETWORK_UUID, null, List.of("P1", "P2").toString(), ElementType.SUBSTATION, null, List.of(150.0, 225.0));
+        succeedingTestForElementsIds(NETWORK_UUID, null, List.of("P0", "P1", "P2", "P3", "P4", "P5", "P6").toString(), ElementType.SUBSTATION, null, null);
+        succeedingTestForElementsIds(NETWORK_UUID, null, List.of("P0", "P1", "P2", "P3", "P4", "P5", "P6").toString(), ElementType.SUBSTATION, null, List.of(24.0, 380.0));
+        succeedingTestForElementsIds(NETWORK_UUID, null, List.of("P0", "P1", "P2").toString(), ElementType.SUBSTATION, null, List.of(150.0, 225.0));
+        succeedingTestForElementsIds(NETWORK_UUID, VARIANT_ID, List.of("P0", "P1", "P2", "P3", "P4", "P5", "P6").toString(), ElementType.SUBSTATION, null, null);
+        succeedingTestForElementsIds(NETWORK_UUID, null, List.of("P0", "P1", "P2", "P3", "P4", "P5", "P6").toString(), ElementType.SUBSTATION, null, List.of(24.0, 380.0));
+        succeedingTestForElementsIds(NETWORK_UUID, null, List.of("P0", "P1", "P2").toString(), ElementType.SUBSTATION, null, List.of(150.0, 225.0));
```
Comment (sBouzols) [inline src/test/java/org/gridsuite/network/map/NetworkMapControllerTest.java:1294]: Maybe we should change to
```suggestion
        succeedingTestForElementsIds(NETWORK_UUID, null, List.of("P0", "P1", "P2", "P3", "P4", "P5", "P6").toString(), ElementType.SUBSTATION, null, null);
        succeedingTestForElementsIds(NETWORK_UUID, null, List.of("P1", "P2", "P3", "P4", "P5", "P6").toString(), ElementType.SUBSTATION, null, List.of(24.0, 380.0));
        succeedingTestForElementsIds(NETWORK_UUID, null, List.of("P1", "P2").toString(), ElementType.SUBSTATION, null, List.of(150.0, 225.0));
        succeedingTestForElementsIds(NETWORK_UUID, VARIANT_ID, List.of("P0", "P1", "P2", "P3", "P4", "P5", "P6").toString(), ElementType.SUBSTATION, null, null);
        succeedingTestForElementsIds(NETWORK_UUID, null, List.of("P1", "P2", "P3", "P4", "P5", "P6").toString(), ElementType.SUBSTATION, null, List.of(24.0, 380.0));
        succeedingTestForElementsIds(NETWORK_UUID, null, List.of("P1", "P2").toString(), ElementType.SUBSTATION, null, List.of(150.0, 225.0));
```

File: src/test/java/org/gridsuite/network/map/NetworkMapControllerTest.java (hunk 1 above)
Comment (souissimai) [reply src/test/java/org/gridsuite/network/map/NetworkMapControllerTest.java:1282]: yes its duplicated, removed

File: src/test/java/org/gridsuite/network/map/NetworkMapControllerTest.java (hunk 3 above)
Comment (souissimai) [reply src/test/java/org/gridsuite/network/map/NetworkMapControllerTest.java]: done

File: src/main/java/org/gridsuite/network/map/NetworkMapService.java [hunk 4]
```diff
@@ -53,7 +53,10 @@ private PreloadingStrategy getPreloadingStrategy(List<String> substationsIds) {
     public List<String> getSubstationsIds(UUID networkUuid, String variantId, List<Double> nominalVoltages) {
         Network network = getNetwork(networkUuid, PreloadingStrategy.COLLECTION, variantId);
         return network.getSubstationStream()
-                .filter(substation -> substation.getVoltageLevelStream().anyMatch(voltageLevel -> nominalVoltages == null || nominalVoltages.contains(voltageLevel.getNominalV())))
+                .filter(substation -> nominalVoltages == null &&
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/network/map/NetworkMapService.java:56]: .filter(substation -> nominalVoltages == null ||
                        substation.getVoltageLevelStream().anyMatch(voltageLevel -> nominalVoltages.contains(voltageLevel.getNominalV())))

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

**Review by sBouzols — APPROVED** [review]
Code Review OK
Tests OK
* substationInfos presents in spreadsheet
* substationId presents in vl creation form

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**